Fixed #87 - Fixes value set query by ensuring the queryable is not prematurely executed.

rustyswayne opened the pull request.

j2jensen commented:
@rustyswayne I don't think the changes you made here actually change any behavior. How did you validate the issue in the first place, and validate that the issue was fixed by this code?

rustyswayne commented:
@j2jensen Before hand ANYTIME a search was executed filtering by name and code system which was ordered by codeCount an exception was thrown indicating that the reader (at various) was already closed.  Your correct, the previous pull to modify the underlying query to accept an array of status codes actually fixed the issue and I had started futzing with things in a new branch, got pulled away, went back and could not reproduce the problem ;-)   I should have just closed the issue rather than doing a pull.

I cleaned things up and added an integration test.
